[infra] detect changes fails on merge-queues

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/19025724039/job/54328970021

```log
2025-11-03T06:20:09.4259815Z Current runner version: '2.329.0'
2025-11-03T06:20:09.4292324Z ##[group]Runner Image Provisioner
2025-11-03T06:20:09.4293179Z Hosted Compute Agent
2025-11-03T06:20:09.4293752Z Version: 20251016.436
2025-11-03T06:20:09.4294471Z Commit: 8ab8ac8bfd662a3739dab9fe09456aba92132568
2025-11-03T06:20:09.4295089Z Build Date: 2025-10-15T20:44:12Z
2025-11-03T06:20:09.4295662Z ##[endgroup]
2025-11-03T06:20:09.4296122Z ##[group]Operating System
2025-11-03T06:20:09.4296662Z Microsoft Windows Server 2025
2025-11-03T06:20:09.4297177Z 10.0.26100
2025-11-03T06:20:09.4297619Z Datacenter
2025-11-03T06:20:09.4298016Z ##[endgroup]
2025-11-03T06:20:09.4298480Z ##[group]Runner Image
2025-11-03T06:20:09.4298974Z Image: windows-2025
2025-11-03T06:20:09.4299401Z Version: 20251021.67.1
2025-11-03T06:20:09.4300361Z Included Software: https://github.com/actions/runner-images/blob/win25/20251021.67/images/windows/Windows2025-Readme.md
2025-11-03T06:20:09.4302134Z Image Release: https://github.com/actions/runner-images/releases/tag/win25%2F20251021.67
2025-11-03T06:20:09.4303025Z ##[endgroup]
2025-11-03T06:20:09.4303968Z ##[group]GITHUB_TOKEN Permissions
2025-11-03T06:20:09.4305886Z Contents: read
2025-11-03T06:20:09.4306382Z Metadata: read
2025-11-03T06:20:09.4306794Z ##[endgroup]
2025-11-03T06:20:09.4308997Z Secret source: Actions
2025-11-03T06:20:09.4309590Z Prepare workflow directory
2025-11-03T06:20:09.4693595Z Prepare all required actions
2025-11-03T06:20:09.4735508Z Getting action download info
2025-11-03T06:20:09.8265520Z Download action repository 'actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8' (SHA:08c6903cd8c0fde910a37f88322edcfb5dd907a8)
2025-11-03T06:20:09.9618714Z Download action repository 'AurorNZ/paths-filter@3b1f3abc3371cca888d8eb03dfa70bc8a9867629' (SHA:3b1f3abc3371cca888d8eb03dfa70bc8a9867629)
2025-11-03T06:20:10.7906417Z Complete job name: detect-changes
2025-11-03T06:20:10.9788377Z ##[group]Run AurorNZ/paths-filter@3b1f3abc3371cca888d8eb03dfa70bc8a9867629
2025-11-03T06:20:10.9791873Z with:
2025-11-03T06:20:10.9842332Z   filters: md: [ '**.md' ]
yml: [ '**.yml', '**.yaml', '.yamllint' ]
build: ['build/**', '.github/**/*.yml', '!.github/workflows/package-*', '**/*.targets', '**/*.props', 'global.json']
shared: ['src/Shared/**', 'test/Shared/**']
contrib-shared-tests: ['test/OpenTelemetry.Contrib.Shared.Tests/**', 'test/OpenTelemetry.Contrib.Shared.Benchmarks/**']
code: ['**.cs', '**.csproj', '.editorconfig']
aottestapp: ['test/OpenTelemetry.AotCompatibility.TestApp/**']
exporter-geneva: ['*/OpenTelemetry.Exporter.Geneva*/**', '!**/*.md']
exporter-influxdb: ['*/OpenTelemetry.Exporter.InfluxDB*/**', '!**/*.md']
exporter-instana: ['*/OpenTelemetry.Exporter.Instana*/**', '!**/*.md']
exporter-onecollector: ['*/OpenTelemetry.Exporter.OneCollector*/**', '!**/*.md']
extensions: ['*/OpenTelemetry.Extensions/**', '*/OpenTelemetry.Extensions.Tests/**', '!**/*.md']
extensions-enrichment: ['*/OpenTelemetry.Extensions.Enrichment/**', '*/OpenTelemetry.Extensions.Enrichment.Tests/**', '!**/*.md']
extensions-enrichment-aspnetcore: ['*/OpenTelemetry.Extensions.Enrichment.AspNetCore*/**', '!**/*.md']
extensions-enrichment-http: ['*/OpenTelemetry.Extensions.Enrichment.Http*/**', '!**/*.md']
instrumentation-aspnet: ['*/OpenTelemetry.Instrumentation.AspNet/**', '*/OpenTelemetry.Instrumentation.AspNet.Tests/**', '*/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule*/**', 'examples/AspNet/**', '!**/*.md']
instrumentation-aspnetcore: ['*/OpenTelemetry.Instrumentation.AspNetCore*/**', '!**/*.md']
instrumentation-aws: ['*/OpenTelemetry.Extensions.AWS*/**', '*/OpenTelemetry.Instrumentation.AWS*/**', '!**/*.md']
instrumentation-cassandra: ['*/OpenTelemetry.Instrumentation.Cassandra*/**', '!**/*.md']
instrumentation-confluentkafka: ['*/OpenTelemetry.Instrumentation.ConfluentKafka*/**', 'examples/kafka/**', '!**/*.md']
instrumentation-elasticsearchclient: ['*/OpenTelemetry.Instrumentation.ElasticsearchClient*/**', '!**/*.md']
instrumentation-entityframeworkcore: ['*/OpenTelemetry.Instrumentation.EntityFrameworkCore*/**', '!**/*.md']
instrumentation-eventcounters: ['*/OpenTelemetry.Instrumentation.EventCounters*/**', 'examples/event-counters/**', '!**/*.md']
instrumentation-grpccore: ['*/OpenTelemetry.Instrumentation.GrpcCore*/**', '!**/*.md']
instrumentation-grpcnetclient: ['*/OpenTelemetry.Instrumentation.GrpcNetClient*/**', '!**/*.md']
instrumentation-hangfire: ['*/OpenTelemetry.Instrumentation.Hangfire*/**', '!**/*.md']
instrumentation-http: ['*/OpenTelemetry.Instrumentation.Http*/**', '!**/*.md']
instrumentation-owin: ['*/OpenTelemetry.Instrumentation.Owin*/**', 'examples/owin/**', '!**/*.md']
instrumentation-process: ['*/OpenTelemetry.Instrumentation.Process*/**', 'examples/process-instrumentation/**', '!**/*.md']
instrumentation-quartz: ['*/OpenTelemetry.Instrumentation.Quartz*/**', '!**/*.md']
instrumentation-runtime: ['*/OpenTelemetry.Instrumentation.Runtime*/**', 'examples/runtime-instrumentation/**', '!**/*.md']
instrumentation-servicefabricremoting: ['*/OpenTelemetry.Instrumentation.ServiceFabricRemoting*/**', '!**/*.md']
instrumentation-sqlclient: ['*/OpenTelemetry.Instrumentation.SqlClient*/**', '!**/*.md']
instrumentation-stackexchangeredis: ['*/OpenTelemetry.Instrumentation.StackExchangeRedis*/**', 'examples/redis/**', '!**/*.md']
instrumentation-wcf: ['*/OpenTelemetry.Instrumentation.Wcf*/**', 'examples/wcf/**', '!**/*.md']
opamp-client: ['*/OpenTelemetry.OpAmp.Client*/**', '!**/*.md']
persistentstorage: ['*/OpenTelemetry.PersistentStorage*/**', '!**/*.md']
resources-aws: ['*/OpenTelemetry.Resources.AWS*/**', '!**/*.md']
resources-azure: ['*/OpenTelemetry.Resources.Azure*/**', '!**/*.md']
resources-container: ['*/OpenTelemetry.Resources.Container*/**', '!**/*.md']
resources-gcp: ['*/OpenTelemetry.Resources.Gcp*/**', '!**/*.md']
resources-host: ['*/OpenTelemetry.Resources.Host*/**', '!**/*.md']
resources-operatingsystem: ['*/OpenTelemetry.Resources.OperatingSystem/**', '*/OpenTelemetry.Resources.OperatingSystem.Tests/**', '!**/*.md']
resources-process: ['*/OpenTelemetry.Resources.Process/**', '*/OpenTelemetry.Resources.Process.Tests/**', '!**/*.md']
resources-processruntime: ['*/OpenTelemetry.Resources.ProcessRuntime/**', '*/OpenTelemetry.Resources.ProcessRuntime.Tests/**', '!**/*.md']
sampler-aws: ['*/OpenTelemetry.Sampler.AWS*/**', '!**/*.md']
semanticconventions: ['*/OpenTelemetry.SemanticConventions*/**', '!**/*.md']

2025-11-03T06:20:10.9885224Z   token: ***
2025-11-03T06:20:10.9886145Z   list-files: none
2025-11-03T06:20:10.9886983Z   initial-fetch-depth: 100
2025-11-03T06:20:10.9889182Z ##[endgroup]
2025-11-03T06:20:11.1777993Z ##[group]Get current git ref
2025-11-03T06:20:11.2879533Z [command]"C:\Program Files\Git\bin\git.exe" branch --show-current
2025-11-03T06:20:11.8472439Z fatal: not a git repository (or any of the parent directories): .git
2025-11-03T06:20:11.8535588Z ##[endgroup]
2025-11-03T06:20:11.8556959Z ##[error]The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
2025-11-03T06:20:11.8788027Z Evaluate and set job outputs
2025-11-03T06:20:11.8801146Z Cleaning up orphan processes
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -29,7 +29,7 @@ jobs:
     steps:
     - name: check out code
       uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
-      if: ${{ github.event_name == 'push' || github.event_name == 'merge_group' }}
+      if: github.event_name == 'push' || github.event_name == 'merge_group'
 
     - uses: AurorNZ/paths-filter@3b1f3abc3371cca888d8eb03dfa70bc8a9867629 # v4.0.0
       id: changes

